tsdb/series_partition.go

```diff
@@ -572,14 +580,14 @@ func (c *SeriesPartitionCompactor) compactIndexTo(index *SeriesIndex, seriesN ui
 				return fmt.Errorf("unexpected series partition log entry flag: %d", flag)
 			}
 
+			// Save max series identifier processed.
+			hdr.MaxSeriesID, hdr.MaxOffset = id, offset
+
```

Comment:
This is another unrelated bug but it caused deleted series to not move the max id/offset during compaction. This is now done before the deletion check.